support-frontend/assets/components/ticker/contributionTicker.jsx

```diff
@@ -45,74 +64,114 @@ export default class ContributionTicker extends Component<PropTypes, StateTypes>
-      progressBarElement.style.transform = `translateX(${percentageTotalAsNegative(totalSoFar, goal)}%)`;
+  tickerType: TickerType;
+  classModifiers: Array<?string>;
+  goalReached: boolean;
+  dataFromServer: DataFromServer;
+
+  increaseTextCounter = () => {
+    const nextCount = this.state.count ? this.state.count + Math.floor(this.state.count / 100) : null;
+
+    if (nextCount && this.dataFromServer.totalContributed && nextCount >= this.dataFromServer.totalContributed) {
+      this.setState({ count: this.dataFromServer.totalContributed });
+    } else if (nextCount) {
+      this.setState({ count: nextCount });
+      window.requestAnimationFrame(this.increaseTextCounter);
     }
   }
 
-  increaseCounter() {
-    const { totalSoFar } = this;
-    this.count += Math.floor(totalSoFar / 100);
+  renderContributedSoFar = () => {
+    if (!this.goalReached && this.state.count) {
+      return (
+        <div className="contributions-landing-ticker__so-far">
+          <div className="contributions-landing-ticker__count">${Math.floor(this.state.count).toLocaleString()}</div>
+          <div className="contributions-landing-ticker__count-label contributions-landing-ticker__label">contributed</div>
+        </div>
+      );
+    }
 
-    if (this.count >= this.totalSoFar) {
-      this.setState({ totalSoFar: this.totalSoFar });
-    } else {
-      this.setState({ totalSoFar: this.count });
-      window.requestAnimationFrame(this.increaseCounter);
+    if (this.goalReached && this.tickerType === 'unlimited') {
+      return (
+        <div className="contributions-landing-ticker__so-far">
+          <div className="contributions-landing-ticker__count">We&#39;ve met our goal &mdash; thank you</div>
+          <div className="contributions-landing-ticker__count-label contributions-landing-ticker__label">Contributions are still being accepted</div>
+        </div>
+      );
     }
+
+    if (this.goalReached && this.tickerType === 'hardstop') {
+      return (
+        <div className="contributions-landing-ticker__so-far">
+          <div className="contributions-landing-ticker__count">We&#39;ve met our goal &mdash; thank you</div>
+        </div>
+      );
+    }
+
+    return (<div className="contributions-landing-ticker__so-far" />);
   }
 
 
   render() {
+    const readyToRender = (this.state && this.state.count && this.state.count !== 0);
+    const allClassModifiers = readyToRender ? this.classModifiers : [...this.classModifiers, 'hidden'];
     const baseClassName = 'contributions-landing-ticker';
-    const wrapperClassName = (this.state && this.state.totalSoFar && this.state.totalSoFar !== 0) ?
-      baseClassName :
-      classNameWithModifiers(baseClassName, ['hidden']);
+    const wrapperClassName = classNameWithModifiers(baseClassName, allClassModifiers);
+    const progressBarAnimation = `translate3d(${percentageToTranslate(this.dataFromServer.totalContributed || 0, this.dataFromServer.goal || 0, this.tickerType)}%, 0, 0)`;
```

Comment:
can we get that brief comment about why we're doing it with translate3d please? any links to references would be helpful too

Reply:
👍 adding this in: 

   /* *************************************************************
      `translate3d` is preferred to `translateX` because it uses the
      gpu to accelerate the animation. This is especially helpful for
      older computers or phones which may not have the CPU to handle
      two animations happening at the same time (the text counter and
      progress bar). Any transitions with `3d` or `z` in the name
      give you this acceleration, even when the `z-axis` is not being
      operated on.
      https://blog.teamtreehouse.com/increase-your-sites-performance-with-hardware-accelerated-css
      https://davidwalsh.name/translate3d
      *************************************************************** */